<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="trajopt_description" default="trajopt_description"/>
  <arg name="plotting" default="true"/>
  <arg name="rviz" default="false"/>
  <arg name="ifopt" default="true"/>
  <arg name="debug" default="false"/>
  <arg name="testing" default="false"/>
  <arg name="method_id" default="1"/>
  <arg name="verbose" default="false"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg trajopt_description)" command="$(find xacro)/xacro --inorder '$(find kortex_description)/robots/urdf/gen3_robotiq_2f_85_tesseract.xacro'" />

  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg trajopt_description)_semantic" textfile="$(find kortex_description)/robots/urdf/gen3_robotiq_2f_85_tesseract.srdf" />

  <group  unless="$(arg testing)">
  <!-- <group ns = "/my_gen3" unless="$(arg testing)"> -->

    <!-- If needed, broadcast static tf for robot root -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 world base_link 100" /> -->
    <rosparam file="$(find feeding_task)/config/feeding_config.yaml" command="load" />

    <node pkg="feeding_task" type="reach_food_item_loop" name="reach_food_item_loop" output="screen">
      <param name="plotting" type="bool" value="$(arg plotting)"/>
      <param name="rviz" type="bool" value="$(arg rviz)"/>
      <param name="ifopt" type="bool" value="$(arg ifopt)"/>
      <param name="debug" type="bool" value="$(arg debug)"/>
      <param name="method_id" type="int" value = "$(arg method_id)"/>
      <param name="verbose" type="bool" value = "$(arg verbose)"/>
    </node>

    <node pkg="anygrasp_generation" type="anygrasp_publisher.py" name="anygrasp_publisher" output="screen" required="true"/>


    <!-- Launch visualization -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="kortex_motion_planning_rviz"
        args="-d $(find kortex_motion_planning)/config/tesseract/kortex_motion_planning.rviz"/>


    <!-- <node name="kortex_motion_execution_node" pkg="kortex_motion_planning" type="kortex_motion_execution_node" output="screen">
    </node> -->
  </group>

</launch>

<!-- <launch>
    <node pkg="anygrasp_generation" type="anygrasp_publisher.py" name="anygrasp_publisher" output="screen" required="true"/>

    <group ns = "/my_gen3">
        <node pkg="feeding_task" name="reach_food_item" type="reach_food_item" output="screen" required="true"/>

    </group>

</launch> -->


